src/settingsholder.h

```diff
@@ -31,6 +33,12 @@ class SettingsHolder final : public QObject {
 #include "settingslist.h"
 #undef SETTING
 
+#define EXPERIMENTAL_FEATURE(experimentId, ...) \
+  Q_PROPERTY(SettingGroup experimentId READ experimentId)
```

Comment:
I should test this, pretty sure it's not working as the type is wrong. It's actually `SettingsGroup*`.

Reply:
Yes, looks like it.

Reply:
Ok, fun times. This is indeed not working and I've spent the whole day debugging it to no success.

I am 99% sure there is a Qt MOC bug here. Some stuff I learned: 

Q_PROPERTY, Q_INVOKABLE, Q_SIGNAL, Q_OBJECT etc. These are _not_ preprocessor macros. See: https://forum.qt.io/topic/79689/the-qt-moc-and-preprocessor

From Qt5 on, it is allowed to use these special MOC macros inside preprocessor macros. Before that, they would get ignored. See: https://bugreports.qt.io/browse/QTBUG-35

Now, the weird thing here is that it looks like we are experiencing is a re-appearance of QTBUG-35. All MOC macros I add inside this specific EXPERIMENTAL_FEATURE definition get ignored. I inspected the moc_ file generated and indeed the generated code expected is not there. If I manually add the experiment settings code, it works fine:

```
// This works!
Q_PROPERTY(SettingGroup* myExperimentalFeature READ myExperimentalFeature
                 WRITE experimentSetter NOTIFY myExperimentalFeatureChanged)
Q_SIGNAL void myExperimentalFeatureChanged();
```

The behaviour is not exactly like QTBUG-35 though. Given we have the SETTING macro definition right below this one and that one has MOC macros inside it that work just fine. I spent today trying to understand the issue in order to report a QTBUG. But I was not successful. So now this has become a 2024 Bea problem 👋

Reply:
I guess what I should have done is punted this to a follow up ticket so as not to block this one 🤔 Leaving this as a note to start the new year doing that.

Reply:
https://mozilla-hub.atlassian.net/browse/VPN-6057